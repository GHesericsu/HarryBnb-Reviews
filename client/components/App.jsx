import React from 'react';
import axios from 'axios';
import styled from 'styled-components';
import MoreReviews from './MoreReviews.jsx';
import OverallStars from './OverallStars.jsx';
import OtherStars from './OtherStars.jsx';
import ReviewList from './ReviewList.jsx';

const Wrapper = styled.div`
  padding: 48px 0px;
  max-width: 1120px;
  display: flex;
  flex-flow: column wrap;
  width: 1120px;
  height: 935px;

`;


class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      ReviewsData: [],
      loaded: false,
    };
    this.getRoomReviews = this.getRoomReviews.bind(this);
  }

  componentDidMount() {
    this.getRoomReviews(3);
  }

  async getRoomReviews(id) {
    try {
      const response = await axios.get(`/rooms/${id}/reviews`);
      this.setState(() => ({
        ReviewsData: response.data,
        loaded: true,
      }));
    } catch (err) {
      console.log(err);
    }
  }


  render() {
    console.log(this.state.ReviewsData);
    return (

      <Wrapper>
        <OverallStars />
        <OtherStars />
        <ReviewList />
        <MoreReviews />
      </Wrapper>

    );
  }
}

export default App;
