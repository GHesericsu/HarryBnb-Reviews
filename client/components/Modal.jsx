import React from 'react';
import styled from 'styled-components';

const ModalBackground = styled.div`
  position:fixed;
  top: 0px;
  left: 0px;
  width: 100%;
  height: 100%;
  background: rgba(34,34,34, 0.7);
  height: 100vh;
  overflow: auto;
  animation-duration: 400ms;
  animation-iteration-count: 1;
  animation-fill-mode: both;
  animation-name: keyframe_15p0df;
  z-index: 2000;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const ModalContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center; 
  z-index: 2000;
  position: fixed;
  padding: 40px;
  
`;

const ModalBox = styled.div`
  width: 1032px;
  max-width: 1032px;
  min-height:300px;
  border-radius: 12px;
  position: relative;
  display: flex;
  flex-direction: column;
  box-shadow: rgba(0, 0, 0, 0.28) 0px 8px 28px;
  animation-duration: 400ms;
  animation-iteration-count: 1;
  animation-fill-mode: both;
  animation-name: keyframe_d37zz3;
  background: rgb(255, 255, 255);
  overflow: auto;
`;

const ExitBox = styled.div`
  top: 24px;
  left: 24px;
  line-height: 20px;
  position: absolute;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 24px;
  width: 24px;
  font-size: 14px;
  color: rgb(72, 72, 72)
`;

const ExitButton = styled.button`
  color: rgb(34, 34, 34);
  cursor: pointer;
  position: relative;
  background: transparent;
  border-color: transparent;
  transition: -ms-transform 0.25s ease 0s, -webkit-transform 0.25s ease 0s, transform 0.25s ease 0s;
  border-radius: 50%;
  display: flex;
  height: 32px;
  width: 32px;
  align-items: center;
  justify-content: center;
  text-align: center;
  &:hover {
    background-color: rgb(240, 242, 242);
  }
  &:before {
    content: "";
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width: 32px;
    height: 32px;
    border-radius: 50%;
  }
`;

const IconSpan = styled.span`
  position: relative;
  text-align: center;
  font-size: 14px;
  box-sizing: border-box;
  top: 2px;
  display: block;
`

const Icon = styled.svg.attrs(() => ({ role: "presentation", viewBox: "0 0 32 32", xmlns: 'http://www.w3.org/2000/svg', }))`
  height: 16px;
  fill: none;
  width: 16px;
  stroke: rgb(34, 34, 34);
  stroke-width: 3;
  overflow: visible;
`

const SlashOne = styled.path.attrs(() => ({ d: "m6 6 20 20" }))`
  
`;

const SlashTwo = styled.path.attrs(() => ({ d: "m26 6-20 20" }))`
  
`;
const TopBox = styled.div`
  flex-grow: 0;
  flex-shrink: 0;
  flex-basis: 72px;
  position: sticky;
  z-index: -1;
  top: 0px;
`;

const MainBox = styled.div`
  background-color: lightgray;
`;

const Modal = ({ handleHideModal, showModal }) => {
  if (showModal === false) {
    return null;
  }
  return (
    <ModalBackground onClick={() => handleHideModal()}>
      <ModalContainer>
        <ModalBox>
          <ExitBox>
            <ExitButton type="button">
              <IconSpan>
                <Icon>
                  <SlashOne />
                  <SlashTwo />
                </Icon>
              </IconSpan>
            </ExitButton>     
          </ExitBox>
          <TopBox>
          Top
          </TopBox>
          <MainBox>
            Main
          </MainBox>
        </ModalBox>
      </ModalContainer>
    </ModalBackground>
  );
};

export default Modal;
