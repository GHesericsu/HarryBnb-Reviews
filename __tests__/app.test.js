import App from '../client/components/App.jsx';
// import Root from '../client/index.jsx';
// import OverallStars from '../client/components/OverallStars.jsx';
import Modal from '../client/components/Modal.jsx';
import data from '../client/components/sampleData.js';
// const data = require('../server/sampleData.js');

describe('App Component', () => {
  test('App component exists', () => {
    const wrapper = mount(<App />);
    const instance = wrapper.instance();
    // console.log('wrapper', wrapper.find(App).children() )
    expect(wrapper.find(App).exists()).toEqual(true);
    // expect(wrapper.find(App).children() > 4).toEqual(true);
    expect(instance).toBeInstanceOf(App);
    wrapper.unmount();
  });

  // xtest('renders correctly', () => {
  //   const wrapper = mount(<App />)
  //   expect(wrapper).toMatchSnapshot();
  //   wrapper.unmount();
  // })
  
  test('Modal should render if showModal state is true', () => {
    const wrapper = mount(<App />);
    wrapper.setState({showModal: true, loaded: true});
    expect(wrapper.find(Modal).exists()).toBe(true);
    wrapper.unmount();
  })
});


// console.log(data.sampleData)