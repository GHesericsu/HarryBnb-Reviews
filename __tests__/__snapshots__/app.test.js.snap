// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `
<App>
  <div>
    <Memo(GlobalStyleComponent)
      modalOpened={false}
    />
    <App__Wrapper>
      <div
        className="App__Wrapper-sc-1lnx15z-0 ijWOKF"
      />
    </App__Wrapper>
  </div>
</App>
`;
